commit a8f170c5611585510033764926b0c40a6901d001
Author: Cameron Hart <cameron.hart@gmail.com>
Date:   Sun Jul 10 23:55:53 2016 +1000

    [rust] Add accessors for MCSubtargetInfo CPU and Feature tables
    
    This is needed for `-C target-cpu=help` and `-C target-feature=help` in rustc

diff --git a/llvm/include/llvm/MC/MCSubtargetInfo.h b/llvm/include/llvm/MC/MCSubtargetInfo.h
index e1f0a86141e3..1e529bf4fb31 100644
--- a/llvm/include/llvm/MC/MCSubtargetInfo.h
+++ b/llvm/include/llvm/MC/MCSubtargetInfo.h
@@ -230,6 +230,14 @@ public:
     return Found != ProcDesc.end() && StringRef(Found->Key) == CPU;
   }
 
+  ArrayRef<SubtargetSubTypeKV> getCPUTable() const {
+    return ProcDesc;
+  }
+
+  ArrayRef<SubtargetFeatureKV> getFeatureTable() const {
+    return ProcFeatures;
+  }
+
   virtual unsigned getHwMode() const { return 0; }
 
   /// Return the cache size in bytes for the given level of cache.
